src/map_core.rs

```diff
@@ -1,406 +1,101 @@
+#![allow(unsafe_code)]
```

Comment:
We need to add a comment to the top-level deny we have for this - to say that we have some sections where it is allowed in the core of the crate.